### DISCUSSION TITLE
`useWatch` to allow use of `defaultValue` (defined in `useForm`) on first render before the field is registered.

### DISCUSSION THREAD

**Discussion by hinkley-dev** [react-hook-form/react-hook-form#8260] [Q&A]
**Is your feature request related to a problem? Please describe.**
I appreciate the docs being updated on the site. There was a point where I would like to use a defaultValue in a useWatch, but on the first render, the value is still undefined. However this is not the case for getValues, so my solution was this:

```
// someVar is rendered and registered in the component it is defined in, but it does have a `defaultValue` set in the `useForm` call. 
  const someVar = useWatch({ control, name: 'someVar', defaultValue: getValues('someVar') })
```

**Describe the solution you'd like**
Could I make a PR, such that a flag could be passed into useWatch such that it would use the getValue (or the real solution would be to use the `defaultValues` on the first render) as the return if the watch would otherwise result in undefined? I am assuming doing that as the default would be a bit more of an overhaul. 
example solution:
```
// someVar is rendered and registered in the component it is defined in, but it does have a `defaultValue` set in the `useForm` call. 
  const someVar = useWatch({ control, name: 'someVar', useDefaultValue: true })
```

**Describe alternatives you've considered**
I have used this work around:
```
  const someVar = useWatch({ control, name: 'someVar', defaultValue: getValues('someVar') })
```
I would also be cool with making this the default as well, but again, I see how that could be a large overhaul

**Additional context**
Here is an issue I used when first looking for a solution to this.
https://github.com/react-hook-form/react-hook-form/issues/3405

Thanks!

**Comment by Moshyfawn** [comment]
What's your use-case again?

If you have `defaultValues` set on `useForm`, the `watch`ed value's gonna be the value from `defaultValues`. If you set `defaultValue` on `Controller`, the initial value for the `watch`ed field is `undefined`.

You're saying, to work around the `undefined` value, you provide the value from `getValues`, but initially RHF doesn't know about your field's `defaultValue` on `Controller`, so it's gonna be `undefined` as well.

If you're able to provide a non-`undefined` value via  `getValues`, it means that your've already set the field's default value via `defaultValues` at `useForm`. So, what the point of setting the field's `defaultValue` via both `defaultValues` at `useForm` and `defaultValue` at `Controller`?

From [RHF docs about `Controller`](https://react-hook-form.com/api/usecontroller/controller):
> You need to **either** set defaultValue at the field-level **or** useForm's defaultValues. undefined is not a valid value.

[An example CSB](https://codesandbox.io/s/purple-waterfall-b6fcer?file=%2Fsrc%2FApp.tsx)

**Comment by hinkley-dev** [comment]
The following has not been the case for us when using `useForm`
> If you have defaultValues set on useForm, the watched value's gonna be the value from defaultValues

I am working on recreating it in a sandbox, so I will let you know when I get that.

But we definitely had a case where. 
```
console.log(watch('someVar')
console.log(getValues('someVar')
```
result in different outputs on the first render where the first one is always `undefined` and the second is the value from `defaultValues`

**Comment by Moshyfawn** [comment]
Huh, yea, a reproduction CSB would be handy

**Comment by hinkley-dev** [comment]
Ok here is an example similar to how I am using it.
https://codesandbox.io/s/react-hook-form-controller-template-forked-umm16t

When an input is typed in for the first name text field, there are two console logs. One is a `watch` and the other is `getValues` on the same field. 

When the text input is changed, allowing the first conditional render, the `watch` value is `undefined` while the `getValues`, on the same field, is using the `defaultValue` .

**Comment by bluebill1049** [comment]
for your usage, you can always do that: `watch("radio", getValues("radio"))`

**Comment by hinkley-dev** [comment]
Sounds good 👍  Do you know why the default values are used for `getValues` and not for `watch` in this case?

**Reply by bluebill1049** [reply]
the `watch` is more strictly bound to the fields.

**Reply by hinkley-dev** [reply]
Got it, thanks. I've really enjoyed using this tool, so big thank you and thanks for the help on this issue!

**Reply by bluebill1049** [reply]
no worries ❤️ that's awesome to hear.
